### PR TITLE
fix: trim whitespace around comma-separate string config vars

### PR DESCRIPTION
**Pull request by trentm**
Fix parsing of comma-separated strings for relevant config vars to allow
whitespace around the commas. E.g.:
    export ELASTIC_APM_TRANSACTION_IGNORE_URLS='/ping, /metrics*'
This affects ARRAY_OPTS and KEY_VALUE_OPTS.

Also, correct the environment variable for setting `transactionIgnoreUrl` from
`ELASTIC_TRANSACTION_IGNORE_URLS` to `ELASTIC_APM_TRANSACTION_IGNORE_URLS`.

Fixes: #1913

### CODE REVIEW

File: lib/config.js
Comment (astorm) [inline lib/config.js:482]: It looks like we're refactoring `normalizeKeyValuePairs` here -- I don't see a lot of test coverage for this method in `lib/config.js` (maybe I'm missing it?).  How would you feel about expanding our test coverage so it's similar in robustness to the _parsing of ARRAY and KEY_VALUE opts_ test?

**Review by astorm — COMMENTED** [review]
If users love anything, it's additional white space.  This is a good idea -- I've got a few questions before approval but no objections to this refinement/bug-fix overall.

File: CHANGELOG.asciidoc
Comment (trentm) [reply CHANGELOG.asciidoc:46]: Yes intentional. I rushed through the last change and didn't update the CHANGELOG with the PR number after I filed the PR.

File: docs/configuration.asciidoc
Comment (trentm) [reply docs/configuration.asciidoc:194]: Yes. I had to go through the Python (hasn't documented transaction_ignore_urls), Ruby (also hasn't doc'd it), to the Java agent which *has* doc'd it to compare.

I'm happy to move this to a separate PR if you prefer. I feel a little bad merging in somewhat unrelated changes to the same commit.

> might be a low hanging spec PR win for the future.

True, it would be good to spec the envvars.

Comment (trentm) [reply docs/configuration.asciidoc:194]: Sergey pointed out https://docs.google.com/spreadsheets/u/1/d/1JJjZotapacA3FkHc2sv_0wiChILi3uKnkwLTjtBmxwU/preview#gid=0 to me earlier today, FWIW.

File: lib/config.js
Comment (trentm) [reply lib/config.js:474]: We don't actually. The guards on config here are somewhat lax. However, I wrote my patch to handle types the same way the previous code was, so it doesn't blow up in *new* ways at least. E.g. this addition to the test suite "works":

```
+    {
+      opts: { disableInstrumentations: [42, {wut: 'wat'}] },
+      expect: { disableInstrumentations: [42, {wut: 'wat'}] }
+    },
+
```

Run output:

```
$ node test/config.js
...
# parsing of ARRAY and KEY_VALUE opts
...
ok 7 opts={"disableInstrumentations":[42,{"wut":"wat"}]} env=undefined -> {"disableInstrumentations":[42,{"wut":"wat"}]}
...
```

File: lib/config.js
Comment (trentm) [reply lib/config.js:482]: This is tested by these lines: https://github.com/elastic/apm-agent-nodejs/pull/1914/files#diff-324e5ae6ed1d64846d943bc35a8a7394f2099af356ca5d9e30413648a3ae124fR923-R938

Is there more that you think should be tested there?

Comment (astorm) [reply lib/config.js:482]: I was thinking of of some direct tests of `normalizeKeyValuePairs` -- but the indirect tests you pointed out should be sufficient

**Review by astorm — APPROVED** [review]
👍

**Pull request merged**
